src/client/pages/OfferNew/Checkout/UserDetailsForm.tsx

```diff
@@ -149,6 +159,26 @@ export const UserDetailsForm: React.FC<Props> = ({
     )
   }
 
+  const handlePhoneChange = (event: React.FormEvent<HTMLInputElement>) => {
+    const { value } = event.currentTarget
+    const phoneNumber = value
+    if (phoneChangeTimeout) {
+      window.clearTimeout(phoneChangeTimeout)
+      setPhoneChangeTimeout(null)
+    }
+    reallySetPhone(phoneNumber)
+    setPhoneChangeTimeout(
+      window.setTimeout(() => {
+        if (phoneValidation.isValidSync(phoneNumber)) {
+          onPhoneChange(phoneNumber)
+          setHasPhoneError(false)
+        } else {
+          setHasPhoneError(true)
+        }
```

Comment:
```suggestion
       // This could be lift up to avoid being created every time
       // You already has access to a regex to validate phone number inside `locale` object
       const phoneValidator = yup.string().matches(locale.phoneNumber.formatRegex)
       const trimmedNumber = phoneNumber.replace(/\s/g,'')
       const isPhoneValid = phoneValidator(trimmedNumber)
       
        if (isPhoneValid) {
          onPhoneChange(phoneNumber)
          setHasPhoneError(false)
        } else {
          setHasPhoneError(true)
        }
```

Additionally, if you end up using mentioned phone regex (`locale.phoneNumber.formatRegex`), you might want to update the placeholder since it doesn't seen to be compliance with it.

I understand that, as the result of QuoteCart API migration, this will be replaced but my concern is to get a simpler version out that's so simple that might not help us at all: i.e If we want to call users to understand why they didn't connect a payment but we don't try to make sure provided number is valid, we might end up with a bunch a numbers that will lead us to nowhere.